### PR TITLE
Deprecated Commands  & converted to ox_lib

### PR DESCRIPTION
**Pull request by TonybynMp4**
## Checklist

- [ ] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [ ] My code fits the style guidelines.
- [ ] My PR fits the contribution guidelines.

### CODE REVIEW

**Comment by TonybynMp4** [comment]
tis gon' break basically all the scripts :)

**Review by FjamZoo — APPROVED** [review]

File: server/commands.lua [hunk 1]
```diff
@@ -1,109 +1,36 @@
...
-    local permissions = QBConfig.Server.Permissions
-    for i=1, #permissions do
-        local permission = permissions[i]
-        ExecuteCommand(('add_ace qbox.%s %s allow'):format(permission, permission))
-    end
-end)
-
--- Register & Refresh Commands
-
-function QBCore.Commands.Add(name, help, arguments, argsrequired, callback, permission, ...)
```
Comment (Manason) [inline server/commands.lua:18]: Why not deprecate the function rather than remove it?

Comment (TonybynMp4) [reply server/commands.lua:18]: It's either deprecating or removing, and at this point idk which one to do honestly

Comment (Manason) [reply server/commands.lua:18]: Removing will be a breaking change. We should deprecate.

**Comment by BerkieBb** [comment]
Add QBCore.Commands.Add back to deprecate it and we'll remove it later and then fix conflicts and linter issues

**Comment by TonybynMp4** [comment]
> Add QBCore.Commands.Add back to deprecate it and we'll remove it later and then fix conflicts and linter issues

Deprecate like the callbacks (bad) or deprecate it the good way

**Comment by BerkieBb** [comment]
That's good, just add `---@deprecated` above the function, and remove the deprecate later comment ad deprecating doesn't mean removing, rather marking as old and soon to be removed

**Comment by Manason** [comment]
> > Add QBCore.Commands.Add back to deprecate it and we'll remove it later and then fix conflicts and linter issues
> 
> Deprecate like the callbacks (bad) or deprecate it the good way

You don't think we should deprecate with warning messages? Too strong?

File: server/commands.lua [hunk 2]
```diff
@@ -1,109 +1,65 @@
...
-            end
-        end
-        TriggerClientEvent('chat:addSuggestions', src, suggestions)
+    for i=1, #arguments do
+        local argument = arguments[i]
+        properties.params[i] = {
+            name = argument.name,
+            help = argument.help,
+            type = argument.type or nil,
+            optional = (argsrequired and false) or (argument.optional ~= nil and argument.optional) or true
```
Comment (Manason) [inline server/commands.lua:17]: argsrequired and false will always be false

File: server/commands.lua [hunk 3]
```diff
@@ -113,42 +69,66 @@ QBCore.Commands.Add('tp', Lang:t("command.tp.help"), { { name = Lang:t("command.
...
-    local permission = tostring(args[2])
+lib.addCommand('addpermission', {
+    help = Lang:t("command.addpermission.help"),
+    params = {
+        {name = Lang:t("command.addpermission.params.id.name"), help = Lang:t("command.addpermission.params.id.help")},
+        {name = Lang:t("command.addpermission.params.permission.name"), help = Lang:t("command.addpermission.params.permission.help")}
+    },
+    restricted = "qbox.god"
+}, function(source, args)
+    local Player = QBCore.Functions.GetPlayer(tonumber(args[Lang:t("command.addpermission.params.id.name")]))
```
Comment (Manason) [inline server/commands.lua:100]: nit; local variable should be camelCase

File: server/commands.lua [hunk 4]
```diff
@@ -179,73 +165,132 @@ QBCore.Commands.Add('closeserver', Lang:t("command.closeserver.help"), {{ name =
...
+        { name = Lang:t("command.setjob.params.job.name"), help = Lang:t("command.setjob.params.job.help") },
+        { name = Lang:t("command.setjob.params.grade.name"), help = Lang:t("command.setjob.params.grade.help"), optional = true }
+    },
+    restricted = "qbox.god"
+}, function(source, args)
+    local Player = QBCore.Functions.GetPlayer(tonumber(args[Lang:t("command.setjob.params.id.name")]))
     if Player then
-        Player.Functions.SetJob(tostring(args[2]), tonumber(args[3]))
+        if args[Lang:t("command.setjob.params.grade.name")] then
+            Player.Functions.SetJob(tostring(args[Lang:t("command.setjob.params.job.name")]), tonumber(args[Lang:t("command.setjob.params.grade.name")]))
```
Comment (Manason) [inline server/commands.lua:248]: Isn't args[Lang:t("command.setjob.params.job.name")] already guaranteed to be a string?

Comment (Manason) [reply server/commands.lua:248]: Can you explain this logic? This used to be args[2] and args[3], but it's args of a locale?

File: server/commands.lua [hunk 5]
```diff
@@ -274,17 +319,23 @@ QBCore.Commands.Add('ooc', Lang:t("command.ooc.help"), {}, false, function(sourc
...
+    params = {
+        { name = Lang:t("command.me.params.message.name"), help = Lang:t("command.me.params.message.help") }
+    },
+    restricted = "qbox.user"
+}, function(source, args)
     if #args < 1 then TriggerClientEvent('QBCore:Notify', source, Lang:t('error.missing_args2'), 'error') return end
     local msg = table.concat(args, ' '):gsub('[~<].-[>~]', '')
     local playerState = Player(source).state
-    playerState:set('me', msg, true)
+    playerState:set('me', 'l\'individu '..msg, true)
```
Comment (Manason) [inline server/commands.lua:337]: individu?

File: server/events.lua [hunk 6]
```diff
@@ -205,25 +205,6 @@ RegisterNetEvent('QBCore:ToggleDuty', function()
...
-    if hasPerm then
-        if QBCore.Commands.List[command].argsrequired and #QBCore.Commands.List[command].arguments ~= 0 and not args[#QBCore.Commands.List[command].arguments] then
-            TriggerClientEvent('QBCore:Notify', src, Lang:t('error.missing_args2'), 'error')
-        else
-            QBCore.Commands.List[command].callback(src, args)
-        end
-    else
-        TriggerClientEvent('QBCore:Notify', src, Lang:t('error.no_access'), 'error')
-    end
-end)
```
Comment (Manason) [inline server/events.lua:225]: Does this need to be deprecated or is it not a breaking change to remove this event?

**Review by Manason — CHANGES_REQUESTED** [review]

File: server/events.lua (hunk 6 above)
Comment (TonybynMp4) [reply server/events.lua:225]: huuummm probably needs to be deprecated i'll figure something out tommorow

File: server/commands.lua (hunk 5 above)
Comment (TonybynMp4) [reply server/commands.lua:337]: > individu?

copy & paste erro mb lol

File: server/commands.lua (hunk 4 above)
Comment (TonybynMp4) [reply server/commands.lua:248]: ox_lib doesn't use args as indexes but as the name you put in the params

File: server/commands.lua (hunk 3 above)
Comment (TonybynMp4) [reply server/commands.lua:100]: so i guess u mean permission should be Permission?

Comment (Manason) [reply server/commands.lua:100]: local Player -> local player

Comment (TonybynMp4) [reply server/commands.lua:100]: Oh yeah makes sense, thought it was using Capital letters for each words

**Comment by TonybynMp4** [comment]
I have hope this PR gets merged one day

**Comment by ChatDisabled** [comment]
one day

**Review by Manason — APPROVED** [review]

**Pull request merged**